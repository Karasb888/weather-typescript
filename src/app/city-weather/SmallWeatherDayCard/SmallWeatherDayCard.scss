.smallWeatherDayCard {
    display: flex;
    flex-direction: column;
    width: 19%;
    padding: 10px;
    box-sizing: border-box;
    background: #3bc7be96;
    border-radius: 10px;
    margin-bottom: 15px;
    box-shadow: 6px 6px 10px 0px rgba(26, 101, 123, 0.75);
}

.date {
    font-family: 'Cooper-Hewitt';
    font-size: 15px;
    padding: 5px 0;
    border-bottom: 1px solid #3196ad;
}

.buttonsContainer {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    padding-bottom: 5px;
    padding-top: 5px;
}

.tabButton {
    line-height: 20px;
    background: #26c2e6;
    font-family: 'Cooper-Hewitt';
    font-weight: 400;
    border: 0.5px solid #2991a94d;
    font-size: 12.5px;
    margin-top: 2px;
    padding: 1px 8px;
    border-radius: 4.5px;
    cursor: pointer;
    transition: 0.15s;
    
    &:hover{
        background: #13b0d4;
    }
}

.activeTabButton {
    background: #13b0d4;
}

.tabButton:not(:last-child) {
    margin-right: 2px;
}

.tabs {
    padding-top: 10px;
    margin-top: auto;
    border-top: 1px solid #3196ad;
}