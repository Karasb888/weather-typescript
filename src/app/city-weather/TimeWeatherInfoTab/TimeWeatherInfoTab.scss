.weatherOption {
    display: flex;
    flex-direction: row;
    align-items: center;
}

.icon {
    padding-right: 5px;
}

.optionName {
    font-size: 15.5px;
    font-family: 'Cooper-Hewitt';
    font-weight: 300;
}

.optionValue {
    font-size: 15px;
    font-family: 'Cooper-Hewitt';
    font-weight: 400;
    justify-self: flex-end;
    margin-left: auto;
}