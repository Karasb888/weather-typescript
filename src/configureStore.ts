import { Store, createStore, applyMiddleware, compose } from 'redux';
import thunkMiddleware from 'redux-thunk';
import { routerMiddleware } from 'connected-react-router';
import { History } from 'history';
import { ApplicationState, createRootReducer } from './store';
import { localStorageMiddlewate } from './store/middlewares/localStorage';

export default function configureStore(history: History, initialState: ApplicationState): Store<ApplicationState> {
    const storeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

    const store = createStore(
        createRootReducer(history),
        initialState,
        storeEnhancers(applyMiddleware(routerMiddleware(history), thunkMiddleware, localStorageMiddlewate))
    );

    return store
};