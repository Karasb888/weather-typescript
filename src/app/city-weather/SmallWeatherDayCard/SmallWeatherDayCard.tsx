import * as React from 'react';
import { NewWeatherDayObj } from './../../../store/weather';
import * as styles from './SmallWeatherDayCard.scss';
import TimeWeatherInfoTab from '../TimeWeatherInfoTab/TimeWeatherInfoTab';

interface Props {
    weather: NewWeatherDayObj;
};

interface State {
    activeTab: number;
}

class SmallWeatherDayCard extends React.Component<Props, State> {
    constructor(props: Props) {
        super(props);

        this.state = {
            activeTab: 1
        };
    }

    public render() {
        return (
            <div className={styles.smallWeatherDayCard}>
                <div>
                    {
                        this.props.weather.timeDayWeather.map((timeWeatherInfo) => {
                        return <div key={timeWeatherInfo.dt}>{timeWeatherInfo}</div>
                        })
                    }
                </div>
                <div>
                    {
                        this.props.weather.timeDayWeather.map((timeWeatherInfo) => {
                            return <TimeWeatherInfoTab timeWeather={timeWeatherInfo} key={timeWeatherInfo.dt}/>
                        })
                    }
                </div>
            </div>
        );
    }
};

export default SmallWeatherDayCard;
