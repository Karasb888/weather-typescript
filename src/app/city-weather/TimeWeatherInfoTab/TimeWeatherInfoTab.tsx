import * as React from 'react';
import { WeatherListObj } from '../../../store/weather';
import TempIcon from '../../icons/TempIcon';
import TempFeelsIcon from '../../icons/TempFeelsIcon';
import CloudsIcon from '../../icons/CloudsIcon';
import WindIcon from '../../icons/WindIcon';
import * as styles from './TimeWeatherInfoTab.scss';

interface Props {
    timeWeather: WeatherListObj;
};

const TimeWeatherInfoTab: React.FC<Props> = ({ timeWeather }) => {
    const iconsProps = {
        height: '25',
        width: '25',
        svgClass: styles.icon
    };

    return (
        <div>
            <div className={styles.weatherOption}>
                <TempIcon
                    {...iconsProps}
                    viewBox="0 0 64 64"
                />
                <div className={styles.optionName}>Temperture:</div>
                <div className={styles.optionValue}>{timeWeather.main.temp}<span className={styles.celsius}>&#8451;</span></div>
            </div>
            <div className={styles.weatherOption}>
                <TempFeelsIcon
                    {...iconsProps}
                    viewBox="0 0 64 64"
                />
                <div className={styles.optionName}>Feels temperture:</div>
                <div className={styles.optionValue}>{timeWeather.main.feels_like}<span className={styles.celsius}>&#8451;</span></div>
            </div>
            <div className={styles.weatherOption}>
                <CloudsIcon
                    {...iconsProps}
                    viewBox="0 -87 463.83425 463"
                />
                <div className={styles.optionName}>Clouds:</div>
                <div className={styles.optionValue}>{timeWeather.clouds.all}%</div>
            </div>
            <div className={styles.weatherOption}>
                <WindIcon
                    {...iconsProps}
                    viewBox="0 0 512 512"
                />
                <div className={styles.optionName}>Wind:</div>
                <div className={styles.optionValue}>{timeWeather.wind.speed}m/s {timeWeather.wind.deg}deg</div>
            </div>
        </div>
    );
};

export default TimeWeatherInfoTab;
