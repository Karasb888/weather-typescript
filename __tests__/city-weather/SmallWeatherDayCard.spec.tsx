import * as React from 'react';
import { mockedStoreKyiv } from '../mockedStoreKyiv.data';
import configureStore from 'redux-mock-store'
import SmallWeatherDayCard from '../../src/app/city-weather/SmallWeatherDayCard/SmallWeatherDayCard';
import { shallow } from 'enzyme';
import TimeWeatherInfoTab from '../../src/app/city-weather/TimeWeatherInfoTab/TimeWeatherInfoTab';

const props = {
    weather: mockedStoreKyiv.weather.weatherCity[0]
};;

describe('++++ <SmallWeatherDayCard /> ++++', () => {
    const wrapper = shallow(<SmallWeatherDayCard {...props} />);
    props.weather.timeDayWeather.forEach((timeWeather) => {
        it('Renders correctly for different tabs', () => {
            wrapper.find(`[data-time=${timeWeather.dt}]`).simulate('click');
            expect(wrapper.state('activeTab')).toEqual(timeWeather.dt);
            expect(wrapper.find(TimeWeatherInfoTab).props()).toEqual({timeWeather});
            expect(wrapper).toMatchSnapshot();
        });
    });
});