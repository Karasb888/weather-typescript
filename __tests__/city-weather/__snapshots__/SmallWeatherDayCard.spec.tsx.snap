// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`++++ <SmallWeatherDayCard /> ++++ Renders correctly for different tabs 1`] = `
<div
  className="smallWeatherDayCard"
>
  <div
    className="date"
  >
    Saturday, February 22, 2020
  </div>
  <div
    className="buttonsContainer"
  >
    <div
      className="tabButton activeTabButton"
      data-time={1582372800}
      key="1582372800"
      onClick={[Function]}
    >
      14
      -00
    </div>
    <div
      className="tabButton "
      data-time={1582383600}
      key="1582383600"
      onClick={[Function]}
    >
      17
      -00
    </div>
    <div
      className="tabButton "
      data-time={1582394400}
      key="1582394400"
      onClick={[Function]}
    >
      20
      -00
    </div>
    <div
      className="tabButton "
      data-time={1582405200}
      key="1582405200"
      onClick={[Function]}
    >
      23
      -00
    </div>
  </div>
  <div
    className="tabs"
  >
    <TimeWeatherInfoTab
      timeWeather={
        Object {
          "clouds": Object {
            "all": 53,
          },
          "dt": 1582372800,
          "dt_txt": "2020-02-22 12:00:00",
          "main": Object {
            "feels_like": 4.86,
            "grnd_level": 1010,
            "humidity": 58,
            "pressure": 1026,
            "sea_level": 1026,
            "temp": 8.8,
            "temp_kf": 2.86,
            "temp_max": 8.8,
            "temp_min": 5.94,
          },
          "sys": Object {
            "pod": "d",
          },
          "weather": Array [
            Object {
              "description": "broken clouds",
              "icon": "04d",
              "id": 803,
              "main": "Clouds",
            },
          ],
          "wind": Object {
            "deg": 217,
            "speed": 3,
          },
        }
      }
    />
  </div>
</div>
`;

exports[`++++ <SmallWeatherDayCard /> ++++ Renders correctly for different tabs 2`] = `
<div
  className="smallWeatherDayCard"
>
  <div
    className="date"
  >
    Saturday, February 22, 2020
  </div>
  <div
    className="buttonsContainer"
  >
    <div
      className="tabButton "
      data-time={1582372800}
      key="1582372800"
      onClick={[Function]}
    >
      14
      -00
    </div>
    <div
      className="tabButton activeTabButton"
      data-time={1582383600}
      key="1582383600"
      onClick={[Function]}
    >
      17
      -00
    </div>
    <div
      className="tabButton "
      data-time={1582394400}
      key="1582394400"
      onClick={[Function]}
    >
      20
      -00
    </div>
    <div
      className="tabButton "
      data-time={1582405200}
      key="1582405200"
      onClick={[Function]}
    >
      23
      -00
    </div>
  </div>
  <div
    className="tabs"
  >
    <TimeWeatherInfoTab
      timeWeather={
        Object {
          "clouds": Object {
            "all": 99,
          },
          "dt": 1582383600,
          "dt_txt": "2020-02-22 15:00:00",
          "main": Object {
            "feels_like": 2.49,
            "grnd_level": 1007,
            "humidity": 62,
            "pressure": 1024,
            "sea_level": 1024,
            "temp": 7.24,
            "temp_kf": 2.14,
            "temp_max": 7.24,
            "temp_min": 5.1,
          },
          "sys": Object {
            "pod": "d",
          },
          "weather": Array [
            Object {
              "description": "overcast clouds",
              "icon": "04d",
              "id": 804,
              "main": "Clouds",
            },
          ],
          "wind": Object {
            "deg": 224,
            "speed": 4.05,
          },
        }
      }
    />
  </div>
</div>
`;

exports[`++++ <SmallWeatherDayCard /> ++++ Renders correctly for different tabs 3`] = `
<div
  className="smallWeatherDayCard"
>
  <div
    className="date"
  >
    Saturday, February 22, 2020
  </div>
  <div
    className="buttonsContainer"
  >
    <div
      className="tabButton "
      data-time={1582372800}
      key="1582372800"
      onClick={[Function]}
    >
      14
      -00
    </div>
    <div
      className="tabButton "
      data-time={1582383600}
      key="1582383600"
      onClick={[Function]}
    >
      17
      -00
    </div>
    <div
      className="tabButton activeTabButton"
      data-time={1582394400}
      key="1582394400"
      onClick={[Function]}
    >
      20
      -00
    </div>
    <div
      className="tabButton "
      data-time={1582405200}
      key="1582405200"
      onClick={[Function]}
    >
      23
      -00
    </div>
  </div>
  <div
    className="tabs"
  >
    <TimeWeatherInfoTab
      timeWeather={
        Object {
          "clouds": Object {
            "all": 97,
          },
          "dt": 1582394400,
          "dt_txt": "2020-02-22 18:00:00",
          "main": Object {
            "feels_like": 0.2,
            "grnd_level": 1007,
            "humidity": 67,
            "pressure": 1023,
            "sea_level": 1023,
            "temp": 5.26,
            "temp_kf": 1.43,
            "temp_max": 5.26,
            "temp_min": 3.83,
          },
          "sys": Object {
            "pod": "n",
          },
          "weather": Array [
            Object {
              "description": "overcast clouds",
              "icon": "04n",
              "id": 804,
              "main": "Clouds",
            },
          ],
          "wind": Object {
            "deg": 201,
            "speed": 4.32,
          },
        }
      }
    />
  </div>
</div>
`;

exports[`++++ <SmallWeatherDayCard /> ++++ Renders correctly for different tabs 4`] = `
<div
  className="smallWeatherDayCard"
>
  <div
    className="date"
  >
    Saturday, February 22, 2020
  </div>
  <div
    className="buttonsContainer"
  >
    <div
      className="tabButton "
      data-time={1582372800}
      key="1582372800"
      onClick={[Function]}
    >
      14
      -00
    </div>
    <div
      className="tabButton "
      data-time={1582383600}
      key="1582383600"
      onClick={[Function]}
    >
      17
      -00
    </div>
    <div
      className="tabButton "
      data-time={1582394400}
      key="1582394400"
      onClick={[Function]}
    >
      20
      -00
    </div>
    <div
      className="tabButton activeTabButton"
      data-time={1582405200}
      key="1582405200"
      onClick={[Function]}
    >
      23
      -00
    </div>
  </div>
  <div
    className="tabs"
  >
    <TimeWeatherInfoTab
      timeWeather={
        Object {
          "clouds": Object {
            "all": 100,
          },
          "dt": 1582405200,
          "dt_txt": "2020-02-22 21:00:00",
          "main": Object {
            "feels_like": -1.76,
            "grnd_level": 1004,
            "humidity": 68,
            "pressure": 1021,
            "sea_level": 1021,
            "temp": 4.22,
            "temp_kf": 0.71,
            "temp_max": 4.22,
            "temp_min": 3.51,
          },
          "sys": Object {
            "pod": "n",
          },
          "weather": Array [
            Object {
              "description": "overcast clouds",
              "icon": "04n",
              "id": 804,
              "main": "Clouds",
            },
          ],
          "wind": Object {
            "deg": 212,
            "speed": 5.48,
          },
        }
      }
    />
  </div>
</div>
`;
